### DISCUSSION TITLE
OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/interface/

### DISCUSSION THREAD

**Discussion by giscus** [runtimevic/OOP-IEC61131-3--Curso-Youtube#12] [General]
# OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/interface/

Curso OOP 61131-3 PLC

https://runtimevic.github.io/OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/interface/

**Comment by anraur** [comment]
Hola Victor.
Para que yo me aclare. ¿Se podría decir, que una interface es como una DUT pero en vez de variables hay métodos y propiedades?
Un saludo

**Comment by runtimevic** [comment]
> Hola Victor. Para que yo me aclare. ¿Se podría decir, que una interface es como una DUT pero en vez de variables hay métodos y propiedades? Un saludo

Hola @anraur , 
- Una interface es como una plantilla la cual esta formada por métodos y propiedades.
- Las intefaces no tienen código en los métodos y propiedades, en la interface solo se declaran los métodos y propiedades...
- La Clase FB que implemente esa interface deberá implementar los métodos y propiedades que tenga la interface.
- La Clase que implementa una interface es como si estableciera un contrato en el cual deberá obligatoriamente implementar los métodos y las propiedades de las interfaces que implemente....
- Como bien has dicho la similitud en los datos es la DUT, que es otra plantilla pero para los datos...
